power_grid_model_io_native_c/power_grid_model_io_native_c/src/vnf_pgm_converter.cpp

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#define PGM_IO_DLL_EXPORTS
+
+#include <power_grid_model_io_native/vnf_converter/vnf_pgm_converter.hpp>
+
+#include "handle.hpp"
+#include <power_grid_model_io_native_c/basics.h>
+#include <power_grid_model_io_native_c/vnf_pgm_converter.h>
+
+#include <power_grid_model/auxiliary/dataset.hpp>
+
+using power_grid_model::ConstDataset;
+
+PGM_IO_VnfConverter* PGM_VNF_create_converter(const PGM_IO_Handle* /*handle*/, char* file_buffer) {
+    auto* converter = new PgmVnfConverter(file_buffer);
+    parse_vnf_file_wrapper(converter);
+    return reinterpret_cast<PGM_IO_VnfConverter*>(converter);
```

Comment:
Since we are in C api and C does not have classes, we are using opaque pointers to hide our classes from C. Since we are communicating between C and C++ we need to use the reinterpret_cast(I found the same usage in PGM and in stackoverflow discussions)